### PR TITLE
WIP: Footer - Sen's updates

### PR DESCRIPTION
**Pull request by MitchAnderson93**
**Task:** 
docs/components-footer--footer

**Features (what's going in):**
>- Standard colours via colour token
>- Basic JS
>- Form.io integration and options (JSON)

**What I am working on today:**
>- Cleaning up everything
>- Making the CSS act correctly within the different themes
>- Looking for errors

**What I still need to do (last):**
>- Making sure what I have done is correct [Design speaking]
>- Any known defects in this PR:

### CODE REVIEW

**Comment by SenChung** [comment]
> --Sen's first pull request (draft)-- Please guide him :)
> 
> **Task:** [FIGMA LINK]
> 
> **Features (what's going in):**
> 
> > * Standard colours
> > * Basic JS
> > * Form.io integration and options (JSON)
> 
> **What I am working on today:**
> 
> > * Cleaning up everything
>> * Making the CSS act correctly within the different themes
>> * Looking for errors
> 
> **What I still need to do (last):**
> 
> > * Look for errors
>> * Making sure what I have done is correct [Design speaking]
> 
> **Any known defects in this PR:**
> 
> >

**Review by stvp-qld — APPROVED** [review]

**Comment by SenChung** [comment]
Fixed mobile container gutter based upon uat.forgov.qld.gov.au/user

**Comment by stvp-qld** [comment]
> uat.forgov.qld.gov.au/user

This is good to merge please @SenChung

**Comment by stvp-qld** [comment]
> I am merging the footer component for its first review.

This is good to merge please @SenChung

**Comment by MitchAnderson93** [comment]
Nice work

**Comment by SenChung** [comment]
Added CSS Tokens and cleaned everything to dynamically work on local css vars + some data structure changes.

**Comment by duttonw** [comment]
Hi @SenChung  are you ready for this be be merged?

File: src/components/bs5/footer/footer.hbs [hunk 1]
```diff
@@ -1,136 +1,111 @@
...
+                            <h3 class="footer-heading">{{feedbackForm.title}}</h3>
+                            <p>{{feedbackForm.content}}</p>
+                        </div>
+                        <div class="col-xs-12 col-md-6 col-lg-12">
+                            <div id="qg-feedback-toggle" class="qg-footer-feedback-wrap">
+                                <a id="btn-footer-feedback" class="btn btn-global-secondary qg-feedback-toggle collapsed" data-bs-toggle="collapse" href="#qg-footer-feedback" role="button" aria-expanded="false" aria-controls="qg-footer-feedback" data-analytics-link-group="qg-feedback">
+                                    {{feedbackForm.btnTitle}}
+                                </a>
+                                <div id="qg-footer-feedback" class="qg-footer-feedback__v2 collapse">
+                                    <script>
```
Comment (duttonw) [inline src/components/bs5/footer/footer.hbs:64]: due to CSP security and inlining of scripts, can all inline script elements be moved to fooder.functions.js.

Its far easier to have it in a js file than inline on a security point of view. see https://content-security-policy.com/examples/allow-inline-script/

File: src/components/bs5/footer/footerForgov.hbs [hunk 2]
```diff
@@ -1,69 +1,93 @@
...
+                            <p>{{ feedbackForm.content }}</p>
+                        </div>
+                        <div class="col-xs-12 col-md-6 col-lg-12">
+                            <div id="qg-feedback-toggle" class="qg-footer-feedback-wrap">
+                                <a id="btn-footer-feedback" class="btn btn-global-secondary qg-feedback-toggle collapsed" data-bs-toggle="collapse" href="#qg-footer-feedback" role="button" aria-expanded="false" aria-controls="qg-footer-feedback" data-analytics-link-group="qg-feedback">
+                                    {{ feedbackForm.btnTitle }}
+                                </a>
+                                <div id="qg-footer-feedback" class="qg-footer-feedback__v2 collapse">
+                                    <script>
+                                        window.formioQldCdnVersion = "v2/v2.x.x-latest";
```
Comment (duttonw) [inline src/components/bs5/footer/footerForgov.hbs:66]: ditto, move the inline script variable into script file is better, and let it use global if not overridden in by an earlier function.

File: src/main.js [hunk 3]
```diff
@@ -61,10 +65,22 @@ window.addEventListener("DOMContentLoaded", () => {
     // Navbar
     initializeNavbar();
 
+    // Footer FormIO Action
+    // Note: This is added here, as there is an issue with breadcrumbShorten() function.
+    //       Will move this once that issue is fixed.
+    const footerFormio = document.getElementById("qg-feedback-toggle");
```
Comment (duttonw) [inline src/main.js:71]: could wrap this in a function like they have done with initBreadcrumb/navbar etc so its self contained.

File: src/main.js [hunk 4]
```diff
@@ -61,10 +65,22 @@ window.addEventListener("DOMContentLoaded", () => {
...
+    if (footerFormio) {
+      document.getElementById("btn-footer-feedback").addEventListener("click", displayFeedbackForm);
+    }
+
     // Breadcrumb
     initBreadcrumb();
 
     // Quick exit
+    window.addEventListener("scroll", positionQuickExit, true);
```
Comment (duttonw) [inline src/main.js:80]: could these functions be moved into the initQuickexit() or positionQuickExit() setup functions?

File: src/main.js [hunk 5]
```diff
@@ -91,23 +105,17 @@ window.addEventListener("DOMContentLoaded", () => {
       link.addEventListener("click", accordionHashLinks);
     });
 
-
     // Video
-    let videoThumbnails = document.querySelectorAll('.video-thumbnail');
+    let videoThumbnails = document.querySelectorAll(".video-thumbnail");
```
Comment (duttonw) [inline src/main.js:109]: I know its not part of this ticket, but this should be extracted into a initVideo() function.

**Review by duttonw — COMMENTED** [review]
Would like to not override hardcoded variables with inline script. Your script should see if a window default is set and use that else go with the internally configured version. This also removed the need for inline javascript vs js file which is better on the CSP front.

File: src/components/bs5/footer/footer.functions.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Init footer formio feedback form
+ *
+ * @memberof module:Footer
+ *
+ * @returns {void}
+ */
+export function displayFeedbackForm() {
+  // eslint-disable-next-line no-undef
+  FormioScript.init();
```
Comment (duttonw) [inline src/components/bs5/footer/footer.functions.js:10]: For technical debt removal:

move formio loader script to data attribute (make it optional)

wrap .init() in a check to see if formioscript is already loaded on the page.
if not existing, check data element on feedback form (if set)
else use hard coded url in this footer.functions.js for formio loader.
load the javascript and fire off the init function.

**Pull request merged**